pypika/enums.py

```diff
@@ -70,7 +70,7 @@ class SetOperation(Enum):
     minus = "MINUS"
 
 
-class DatePart(Enum):
+class DatePart(str, Enum):
```

Comment:
Why have you made this inherit from str?

Reply:
Good catch. I thought there was a reason for it initially, but not anymore.

PR updated.

Reply:
Oh sorry no. The tests are failing without. Let me see.

Reply:
So, it is the reason I initially stated in the commit message. Here is an excerpt:

> In order to fix it, `date_part` is wrapped into a `LiteralValue` to prevent
> quoting. To allow the `DatePart` enum to work also, it is derived from `str` as
> well as `Enum`, thus inheriting the required properties.

Here is a little more explanation:
Before Enum values were quoted properly, the `Extract` function was working fine, but only because `DatePart.year` for example wasn't quoted and stayed `YEAR` instead of `'YEAR'`. The same thing happened if you had called `Extract` with a string literal, instead of one of the Enum members (like `Extract("YEAR")`). Calling it like that produced invalid SQL but wasn't tested, instead relied on the old Enum behavior. To get around the issue, I wrap `date_part` in `LiteralValue` which only accepts strings, so it would fail with `DatePart.year`. Now strings worked, but the old behavior using the Enum doesn't. Here comes the `class DatePart(str, Enum)` making it work with enum members (old behavior) and strings (new behavior).